addons/web/static/src/views/fields/translation_dialog.js

```diff
@@ -15,53 +16,53 @@ export class TranslationDialog extends Component {
         this.orm = useService("orm");
         this.user = useService("user");
 
-        this.terms = [];
-        this.updatedTerms = {};
+        this.state = useState({ terms: [] });
 
         onWillStart(async () => {
-            const languages = await loadLanguages(this.orm);
-            const [translations, context] = await this.loadTranslations(languages);
+            const [translations, context] = await this.loadTranslations();
             let id = 1;
             translations.forEach((t) => (t.id = id++));
-            this.props.isText = context.translation_type === "text";
-            this.props.showSource = context.translation_show_source;
+            this.props.languages = await loadLanguages(this.orm);
+            if (!this.props.languages.find((l) => l[0] === "en_US")) {
+                this.props.languages.push(["en_US", _t("Source Value")]);
+            }
```

Comment:
Let's try to avoid this props mutation. For this use case, maybe you can include a `languages` field in the `state` and populate it here in onWillStart together with the `terms`.

Reply:
The `languages` is never changed after the `onWillStart`. For the xml template, the `languages` is static, why do we have to put it in the `state`?

Reply:
True true. Not in state. But put it in the instance itself. Something like this can work.
```js
this.languages = [...this.props.languages];
if (!this.languages.find((l) => l[0] === "en_US")) {
    this.languages.push(["en_US", _t("Source Value")]);
}
```
The idea is to avoid direct mutation of the props.